app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -9,12 +9,10 @@ def command
   end
 
   def registration_args
-    args = registration_params.except(*ignored_query_args)
-    args['setup_insights'] = registration_params['setup_insights']
-    args['setup_remote_execution'] = registration_params['setup_remote_execution']
-
-    args.delete_if { |_, v| v.blank? }
-        .permit!
+    registration_params.except(*ignored_query_args)
+                       .transform_values! { |v| v == false ? v.to_s : v }
```

Comment:
Few thoughts,
1. Does it make sense to delete blank items before transforming those?
2. In test I can see `update_packages` has true value however `transform_values!` only converting `false` to string. I think we also need to typecast `true` to string?

Reply:
1) That's what we want to prevent. `false.blank?` returns `true`, so it would be filtered out as well, but we want to keep it, hence the conversion to string. `true.blank?` is `false` so it doesn't get filtered out.
2) The answer here is the same

Reply:
Looks good then!